### PR TITLE
UI texture atlas slice shader

### PR DESCRIPTION
**Pull request by ickshonpe**
# Objective

Fixes https://github.com/bevyengine/bevy/issues/14183

## Solution

Reimplement the UI texture atlas slicer using a shader. 

The problems with #14183 could be fixed more simply by hacking around with the coordinates and scaling but that way is very fragile and might get broken again the next time we make changes to the layout calculations. A shader based solution is more robust, it's impossible for gaps to appear between the image slices with these changes as we're only drawing a single quad. 

I've not tried any benchmarks yet but it should much more efficient as well, in the worst cases even hundreds or thousands of times faster.

Maybe could have used the UiMaterialPipeline. I wrote the shader first and used fat vertices and then realised it wouldn't work that way with a UiMaterial. If it's rewritten it so it puts all the slice geometry in uniform buffer, then it might work? Adding the uniform buffer would probably make the shader more complicated though, so don't know if it's even worth it. Instancing is another alternative.

## Testing
The examples are working and it seems to match the old API correctly but I've not used the texture atlas slicing API for anything before, I reviewed the PR but that was back in January.

Needs a review by someone who knows the rendering pipeline and wgsl really well because I don't really have any idea what I'm doing.

### CODE REVIEW

File: assets/textures/texture-slice-test-texture.png
Comment (mockersf) [inline assets/textures/texture-slice-test-texture.png:1]: is this file needed?

**Review by mockersf — APPROVED** [review]
Couldn't find issues reading the code, and it works, but it wasn't a review in depth.

Would the same change be needed for slices on spites?

File: assets/textures/texture-slice-test-texture.png
Comment (ickshonpe) [reply assets/textures/texture-slice-test-texture.png:1]: Nope, I was just using it to test the side and center tiling. I'll remove it.

**Review by alice-i-cecile — APPROVED** [review]
Using a shader for this is reasonable, and I'm happy with the code quality.

File: crates/bevy_ui/src/texture_slice.rs [hunk 1]
```diff
@@ -1,213 +0,0 @@
...
-        let mut flip = Vec2::new(1.0, -1.0);
-        let [mut flip_x, mut flip_y] = [false; 2];
-        if image.flip_x {
-            flip.x *= -1.0;
-            flip_x = true;
-        }
-        if image.flip_y {
-            flip.y *= -1.0;
-            flip_y = true;
-        }
```
Comment (ManevilleF) [inline crates/bevy_ui/src/texture_slice.rs:50]: @ickshonpe thanks for the PR, using a shader is much better than CPU bound logic.
I don't see an equivalent to this flip image logic anywhere in the new code, so it might be missing from the migration

Comment (ickshonpe) [reply crates/bevy_ui/src/texture_slice.rs:50]: Hi felix, yes you're right, I took image flipping out temporarily to make things easier and meant to add it back in at the end but completely forgot about it :sweat:. I think it must be really rare to combine nine slicing with flipping though so it's probably not too critical and shouldn't be difficult to add support in a followup pr.
